src/pubtools/_pulp/tasks/delete.py

```diff
@@ -78,10 +78,19 @@ def add_args(self):
 
         self.parser.add_argument(
             "--repo",
-            help="remove content from these comma-seperated repositories ",
+            help="remove content from these comma-seperated repositories. "
+                 "This argument is mutually exclusive with --all-repos.",
             type=str,
             action=SplitAndExtend,
             split_on=",",
+            default=[]
+        )
+
+        self.parser.add_argument(
+            "--all-repos",
```

Comment:
I see that this is aligned with what the JIRA asked for, but I personally feel making it support `--repo *` might have been better than `--all-repos`.

The main reason for this is that there are several "layers" here (pub client => hub => worker => pubtools-pulp), and the value of "*" could be simply passed through the layers the same as any other repo ID and then interpreted by the final command as "every repo".  All the related complexity would live only in a single layer.

With `--all-repos` and `--repo` being separate and mutually exclusive options though, it seems more likely that multiple layers are going to have to learn about the new option and add new logic to deal with it.

Reply:
Additionally, per later thread about "special" all-rpm-content repos:

If this were implemented as `--repo *` with `*` meaning "all customer repos containing the content", then the user has the choice of passing something like `--repo * --repo all-rpm-content-ab --repo all-rpm-content-cd [...]` to set up a single task which removes files from all customer repos *and also* all-rpm-content repos, if that's what they want to do.

This seems to be impossible if we make `--repo` and `--all-repos` two mutually exclusive options (or potentially leads us to more complexity, like adding a third boolean option to toggle whether `--all-repos` includes special repos).